### PR TITLE
feat(evaluation): multi-value array property support for non-set operators

### PR DESCRIPTION
**Pull request by kyeh-amp**
## Summary

Non-set operators (`is`, `is not`, `contains`, `does not contain`, `greater`, `less`, regex, version) now match array-valued user properties element-wise, returning true if **any** element satisfies the predicate. Previously, array values were stringified via `coerceString` and compared as a single blob, diverging from Amplitude analytics/charts behavior.

Applies to both native PHP arrays and JSON-array-string properties (e.g. `'["a","b"]'`). Set operators retain their existing semantics. Boolean short-circuit and scalar-string paths are unchanged.

## Changes

- `matchCondition` restructured so `coerceStringArray` is called for every non-null, non-boolean value. New `matchStringsNonSet` helper delegates to `matchString` per element.
- `coerceStringArray` gains a `[`-prefix pre-check (via `strncmp`, PHP 7.4-safe) so scalar strings short-circuit before `json_decode`.
- `coerceStringArray` now correctly preserves `"0"` and `""` elements — it used to drop them via default `array_filter` truthiness, which contradicted the spec's "filter nulls only" contract.
- Unreachable `try/catch` around `json_decode` removed (PHP returns `null` on parse failure; the existing `is_array` check already handles it).

## Tests

- Unit test harness + 16 cases in `EvaluationEngineTest::testMultiValueArrayPropertyMatching` covering scalars, JSON-array strings, PHP collections, malformed JSON, empty arrays, leading-whitespace strings, and falsy-string elements across both set and non-set operators.
- 7 new integration tests in `EvaluateIntegrationTest` exercising the array-property flags on the shared evaluation deployment.
- Integration deployment key consolidated to a single superset key per the cross-SDK spec.
- Full suite: 154 tests / 2522 assertions passing; phpstan clean.

## Test plan

- [ ] `./vendor/bin/phpunit --no-coverage`
- [ ] `./vendor/bin/phpstan analyse`
- [ ] Spot-check: flag with `is` operator against a user property set to a PHP array — should now match any element (previously required `set` operator or JSON-array string contortions)

---

> [!NOTE]
> **Medium Risk**
> Changes core flag targeting semantics: non-set operators now evaluate array/JSON-array user properties element-wise, which can alter rollout/segment matches for existing flags. Risk is mitigated by added unit/integration coverage but still impacts evaluation behavior broadly.
> 
> **Overview**
> **Adds multi-value property matching for non-set operators.** `matchCondition` now attempts to coerce non-null, non-boolean properties into string arrays and, for non-set operators (e.g. `is`, `contains`, comparisons, regex, version), returns true if *any* element matches via a new `matchStringsNonSet` helper; set operators keep existing semantics.
> 
> **Tightens array coercion behavior.** `coerceStringArray` now fast-paths non-array strings that don’t start with `[` before attempting `json_decode`, removes the unused `try/catch`, and filters out only `null` elements (preserving falsy strings like `"0"`/`""`) via `filterNonNullStrings`.
> 
> **Updates tests.** Adds comprehensive unit coverage for scalar/array/JSON-array cases and edge conditions, and extends integration tests for array/JSON-array targeting; also updates the integration deployment key used to fetch test flags.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 2045fa9037d3e572a2cc37190c025d8033964f61. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: src/EvaluationCore/EvaluationEngine.php [hunk 1]
```diff
@@ -300,6 +300,22 @@ private function matchString(string $propValue, string $op, array $filterValues)
...
+     */
+    private function matchStringsNonSet(array $propValues, string $op, array $filterValues): bool
+    {
+        foreach ($propValues as $propValue) {
+            if ($this->matchString($propValue, $op, $filterValues)) {
+                return true;
+            }
+        }
+        return false;
+    }
```
[the remark this answers is not in the document]
Comment (vaibhav-jain-exp) [reply src/EvaluationCore/EvaluationEngine.php:317]: Is this valid?

[the remark this answers is not in the document]
Comment (kyeh-amp) [reply src/EvaluationCore/EvaluationEngine.php:317]: It's a valid assertion of the logic, but the relevance is off. In the end the implementation mirrors the observed filtering logic in charts/analytics.
